asn1/Stream.asn

```diff
@@ -0,0 +1,177 @@
+Stream
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt96,
+    UInt128,
+    VarUInt,
+    VarBytes
+    FROM GenericTypes
+
+    Address
+    FROM InterledgerTypes
+;
+
+StreamEncryptionEnvelope ::= SEQUENCE {
+    -- Random nonce
+    initializationVector UInt96,
+
+    -- Authentication tag (output of AES-GCM encryption)
+    authenticationTag UInt128,
+
+    -- Encrypted STREAM Packet
+    -- Note: This is NOT encoded as a variable length (length-prefixed)
+    -- octet string. The size is constrained by the length of the data
+    -- field in the InterledgerPrepare, InterledgerFulfill, or InterledgerReject
+    -- packet containing this StreamEncryptionEnvelope
+    cipherText OCTET STRING (SIZE (0..32739))
+}
+
+StreamPacket ::= SEQUENCE {
+  -- Always 1 for now
+  version UInt8,
+
+  -- ILP Packet type this STREAM packet must be carried upon
+  ilpPacketType UInt8,
+
+  -- Packet ID
+  sequence VarUInt,
+
+  -- If attached to a Prepare packet, minimum amount that the receiver should accept
+  -- If attached to a Fulfill or Reject packet, amount that arrived at the receiver
+  prepareAmount VarUInt,
```

Comment:
Is this not UInt64 to save space?

Reply:
My only thought was that this will always correspond to an amount in an ILP packet but I think you are right that saving bytes should be the priority.